module/userExtension/helper.js

```diff
@@ -0,0 +1,257 @@
+    * @name  getProfile
+    * @param {String}  userId - Logged in user details.
+    * @param {String} token - user access token.
+    * @returns {json} Response consists of user profile details.
+    */
+    static getProfile(userId, token) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let queryObject = {
+                    userId: userId,
+                    status: constants.common.ACTIVE
+                }
+                let profileData = await sunbirdService.getUserProfile(token, userId);
+                let roles;
+                profileData = JSON.parse(profileData);
+
+                if (profileData.responseCode != constants.common.RESPONSE_OK) {
+                    
+                    reject({
+                        status: httpStatusCode.bad_request.status,
+                        message: constants.apiResponses.USER_PROFILE_NOT_FOUND
+                      })
+                }
+
+                if (profileData.result &&
+                    profileData.result.response
+                ) {
+                    if (profileData.result.response.roles) {
+                        roles = profileData.result.response.roles;
+                    }
+                    await Promise.all(profileData.result.response.organisations.map(orgInfo => {
+                        if (orgInfo.roles) {
+                            roles.push(...orgInfo.roles);
+                        }
+                    }))
+                }
+
+
+                if (roles) {
+                    roles = Array.from(new Set(roles));
+                }
+
+                let userExtDocument = await database.models.userExtension.findOne(
+                    queryObject,
+                    {
+                        updatedBy: 0,
+                        createdBy: 0,
+                        createdAt: 0,
+                        updatedAt: 0,
+                        status: 0,
+                        deleted: 0,
+                        userId: 0,
+                        "__v": 0
+                    }
+                ).lean();
+
+                if (userExtDocument) {
+                    userExtDocument.roles = userExtDocument.roles.map(role => role.code)
+                    if (roles) {
+                        userExtDocument.roles.push(...roles);
+                    }
+                }
+                return resolve(userExtDocument);
+
+            } catch (error) {
+                return reject(error);
+            }
+        })
+
+    }
+
+
+    /**
+ * create a new user.
+ * @method
+ * @name  create
+ * @param {String} request - consist of new user details
+ * @param {String} token - user access token
+ * @param {String} userId  - logged in userId.
+ * @returns {json} Response consists of created user details.
+ */
+    static create(request, token, userId) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (request.organisation &&
```

Comment:
@rakeshSgr - If validator is there, then no need of condition here right?

Reply:
resolved